node/authorship/src/tests.rs

```diff
@@ -222,208 +241,172 @@ pub(crate) fn init_logger() {
-    init_logger();
-    gear_runtime_interface::sandbox_init();
+    // Import last block
+    block_on(client.import(BlockOrigin::Own, proposal.block.clone())).unwrap();
 
-    init!(client, backend, txpool, spawner, genesis_hash);
+    assert_eq!(client.info().best_hash, proposal.block.hash());
 
-    let extrinsics = sign_extrinsics(
-        checked_extrinsics(1, bob(), 0_u32, || CallBuilder::noop().build()),
-        VERSION.spec_version,
-        VERSION.transaction_version,
-        genesis_hash,
-    );
+    proposal
+}
 
-    submit_txs!(client, txpool, BlockId::number(0), extrinsics);
-    assert_eq!(txpool.ready().count(), 1);
+fn submit_and_maintain<A>(client: Arc<TestClient>, txpool: Arc<A>, extrinsics: Vec<OpaqueExtrinsic>)
+where
+    A: MaintainedTransactionPool<Block = TestBlock> + 'static,
+{
+    let block_id = BlockId::Number(client.info().best_number);
+    let hash = client.info().best_hash;
 
-    propose_block!(
-        client,
-        backend,
-        txpool,
-        spawner,
-        best_hash,
-        BlockId::number(0),
-        Box::new(time::Instant::now),
-        Timestamp::current(),
-        1500_u64,
-        None,
-        proposal
-    );
-    // then
-    // block should have exactly 3 txs: an inherent (timestamp), a normal and a mandatory one
-    assert_eq!(proposal.block.extrinsics().len(), 3);
+    block_on(txpool.submit_at(&block_id, SOURCE, extrinsics)).unwrap();
+    block_on(txpool.maintain(chain_event(
+        client.expect_header(hash).expect("there should be header"),
+    )));
 }
 
+type TestCase = Box<dyn Fn() + Send + 'static>;
+
 #[test]
-fn proposed_storage_changes_match_execute_block_storage_changes() {
+fn run_all_tests() {
     init_logger();
     gear_runtime_interface::sandbox_init();
 
-    init!(client, backend, txpool, spawner, genesis_hash);
+    use basic_tests::*;
+
+    let tests = vec![
+        Box::new(custom_extrinsic_is_placed_in_each_block) as TestCase,
+        Box::new(queue_remains_intact_if_processing_fails) as TestCase,
+        Box::new(block_max_gas_works) as TestCase,
+        Box::new(terminal_extrinsic_discarded_from_txpool) as TestCase,
+        Box::new(block_builder_cloned_ok) as TestCase,
+        Box::new(proposal_timing_consistent) as TestCase,
+        Box::new(should_cease_building_block_when_deadline_is_reached) as TestCase,
+        Box::new(should_not_panic_when_deadline_is_reached) as TestCase,
+        Box::new(proposed_storage_changes_should_match_execute_block_storage_changes) as TestCase,
+        Box::new(should_not_remove_invalid_transactions_when_skipping) as TestCase,
+        Box::new(should_cease_building_block_when_block_limit_is_reached) as TestCase,
+        Box::new(should_keep_adding_transactions_after_exhausts_resources_before_soft_deadline)
+            as TestCase,
+        Box::new(should_only_skip_up_to_some_limit_after_soft_deadline) as TestCase,
+    ];
+
+    let handles: Vec<_> = tests
+        .into_iter()
```

Comment:
you can simply use par_iter from rayon

Reply:
Have tried it. Somehow rayon stumbles upon some data race, apparently - it's not even able to complete the test. Even with half of the test cases it is already too slow.
The std::thread, on the other hand, is consistent and almost doesn't depend on how many tests are run in parallel.

Will look into it again though because it's really tempting - the code is more elegant that way due to higher level parallelisation primitives